modules/core/src/main/java/org/apache/ignite/internal/cdc/ChangeDataCaptureFileLockHolder.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.cdc;
+
+import java.lang.management.ManagementFactory;
+import java.util.function.Supplier;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.processors.cache.persistence.FileLockHolder;
+
+/**
+ * Lock file holder for CDC application.
+ *
+ * @see ChangeDataCapture
+ * @see ChangeDataCaptureConsumerState
+ */
+public class ChangeDataCaptureFileLockHolder extends FileLockHolder {
+    /** Consumer ID. */
+    private final Supplier<String> consumerId;
```

Comment:
any reason to have it Supplier, not just String?

Reply:
Nope.